### DISCUSSION TITLE
Rootmodel vs type adapter in serializing a pydantic dataclass to json

### DISCUSSION THREAD

**Discussion by david-gang** [pydantic/pydantic#8985] [Question]
Given a pydantic dataclass there are two ways to serialize to json

1.  through type adapter
2. through root model

This is demonstrated in the code below.
From clarity it looks like type adapter is nicer. 

On the other hand the official documentation endorses usage of rootmodel: https://docs.pydantic.dev/latest/concepts/dataclasses/#json-dumping

So when should i use each approach ?

```
from pydantic.dataclasses import dataclass
from pydantic import TypeAdapter, RootModel

@dataclass
class Person:
    name: str
    age: int


adapter = TypeAdapter(Person)

hugo = Person(name="Hugo", age=13)
print(adapter.dump_json(hugo))

print(RootModel[Person](hugo).model_dump_json())
```

**Comment by illeatmyhat** [comment]
That link now endorses the use of `TypeAdapter` which is confusing. I don't see any obvious benefit to `TypeAdapter`
It's so complicated that PyCharm often type hints incorrectly. `RootModel` has the same shape as `BaseModel` and doesn't have any type hinting issues.
